apps/root/src/hooks/useTransactionsHistory.tsx

```diff
@@ -54,22 +51,35 @@ function useTransactionsHistory(): {
         storedWallets.map(({ address }) => address)
       );
 
+      const nonIndexedTransactions = localTransactions.filter(
+        (tx) =>
+          !resolvedEvents.find(
+            (serviceEvent) =>
+              serviceEvent.tx.chainId === tx.chainId && serviceEvent.tx.txHash.toLowerCase() === tx.hash.toLowerCase()
+          )
+      );
+      console.log('nonIndexedTransactions', nonIndexedTransactions);
       const nonIndexedEvents = transformNonIndexedEvents({
         events: nonIndexedTransactions,
         userWallets: storedWallets.map(({ address }) => address),
         tokenList,
         nativePrices,
       });
 
-      resolvedEvents.unshift(...nonIndexedEvents);
+      nonIndexedEvents.forEach((tx) => {
+        const insertionIndex = sortedIndexBy(resolvedEvents, tx, (storedEvent) => -storedEvent.tx.timestamp);
+        resolvedEvents.splice(insertionIndex, 0, tx);
+      });
 
       // This set parsed event is actually killing perf, making this hook re-render a thousand times infinitely
       // Haven't figure out just why this happens, as of right now this is a PATCH for it but not a definitive solution
       if (!isEqual(parsedEvents, resolvedEvents)) {
         setParsedEvents(resolvedEvents);
       }
 
-      if (indexing) dispatch(cleanTransactions({ indexing }));
+      const indexedTransactionsSummary = historyEvents.map(({ tx: { chainId, txHash } }) => ({ chainId, txHash }));
+
+      if (!!localTransactions.length) dispatch(cleanTransactions({ indexedTransactions: indexedTransactionsSummary }));
```

Comment:
this can be a really big dispatch, what about just sending the last blockNumbers of the last transactions of each chain?

Reply:
This would be a large list eventually, but each indexer unit has it's own last block number. We could do this calculation in the hook and send only the transactions to clean. We are kind of doing that already with nonIndexedTransactions

Reply:
Yup that works for me, and we only need to send the chainId and txHash :)